lib/ansible/galaxy/dependency_resolution/reporters.py

```diff
@@ -5,16 +5,103 @@
 
 from __future__ import annotations
 
+import typing as t
+from collections import defaultdict
+from contextlib import suppress as suppress_exceptions
+
 try:
     from resolvelib import BaseReporter
 except ImportError:
     class BaseReporter:  # type: ignore[no-redef]
         pass
 
+try:
+    from resolvelib.resolvers import Criterion
+except ImportError:
```

Comment:
Can we document the version ranges for why we need to try and import from multiple locations. Will help to determine when we can remove it if we move onto a new minimum version that no longer needs this.

Reply:
Brilliant idea! Will do.

Reply:
So I've re-evaluated this and concluded that the nested import in the fallback branch wasn't actually necessary, and I just got confused when they were moving things around.

I've deleted it in the following commit, making adding the requested comments unnecessary. Marking this thread as resolved.